app/javascript/forms/provider-forms/cloud-provider-form/index.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState } from 'react';
+import { componentTypes, validatorTypes } from '@data-driven-forms/react-form-renderer';
+
+import { API } from '../../../http_api';
+import MiqFormRenderer from '../../data-driven-form';
+
+const loadProviderServerZones = () =>
+  API.get('/api/zones?expand=resources&attributes=id,name,visible&filter[]=visible!=false&sort_by=name')
+    .then(({ resources }) => resources.map(({ name, id }) => ({ value: id, label: name })));
```

Comment:
The `zone` needs to be submitted with its name:
```suggestion
    .then(({ resources }) => resources.map(({ name }) => ({ value: name, label: name })));
```